kotlintest-extensions/kotlintest-extensions-spring/src/main/kotlin/io/kotlintest/spring/SpringListener.kt

```diff
@@ -1,22 +1,50 @@
 package io.kotlintest.spring
 
 import io.kotlintest.Spec
+import io.kotlintest.TestCase
+import io.kotlintest.TestResult
 import io.kotlintest.extensions.ConstructorExtension
 import io.kotlintest.extensions.TestListener
+import io.kotlintest.extensions.TopLevelTest
 import org.springframework.beans.factory.config.AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR
 import org.springframework.test.context.TestContextManager
 import kotlin.reflect.KClass
 import kotlin.reflect.full.primaryConstructor
 
 object SpringListener : TestListener {
 
+  // Necessary to keep specs and contexts in a map because this Listener
+  private val testContexts = mutableMapOf<Spec, TestContextManager>()
+
+  override fun beforeSpecClass(spec: Spec, tests: List<TopLevelTest>) {
+    testContexts[spec] = TestContextManager(spec.javaClass)
+  }
+
   override fun beforeSpec(spec: Spec) {
-    try {
-      TestContextManager(spec.javaClass).prepareTestInstance(spec)
-    } catch (t: Throwable) {
-      t.printStackTrace()
-    }
+    spec.testContext.beforeTestClass()
+  }
+
+  override fun beforeTest(testCase: TestCase) {
+    testCase.spec.testContext.beforeTestMethod(testCase.spec, method)
+    testCase.spec.testContext.prepareTestInstance(testCase.spec)
+    testCase.spec.testContext.beforeTestExecution(testCase.spec, method)
+
   }
+
+  override fun afterTest(testCase: TestCase, result: TestResult) {
+    testCase.spec.testContext.afterTestMethod(testCase.spec, method, null as Throwable?)
+  }
+
+  override fun afterSpec(spec: Spec) {
+    spec.testContext.afterTestClass()
+  }
+
+  private val Spec.testContext: TestContextManager
+    get() = testContexts.getValue(this)
+
+  // We don't run methods, but we need to pass one to TestContextManager, so we'll pass any.
+  private val method = SpringListener::class.java.methods[0]
```

Comment:
Should we pick a method we always know will be accessible.  [0] might pick a method that's private or something. I'd suggest using hashcode

Reply:
Good one. I'll make the change